Python/Product/Analysis/Analyzer/DDG.cs

```diff
@@ -430,30 +430,28 @@ internal class DDG : PythonWalker {
 
             var candidates = ModuleResolver.ResolvePotentialModuleNames(_unit.ProjectEntry, modName, forceAbsolute).ToArray();
             foreach (var name in candidates) {
-                if (ProjectState.Modules.TryImport(name, out moduleRef)) {
-                    _unit.DeclaringModule.AddModuleReference(moduleRef);
-                    return true;
-                }
-            }
-
-            foreach (var name in candidates) {
-                moduleRef = null;
-                foreach (var part in ModulePath.GetParents(name, includeFullName: true)) {
-                    if (ProjectState.Modules.TryImport(part, out var mref)) {
-                        moduleRef = mref;
-                        if (part.Length < name.Length) {
-                            moduleRef.Module?.Imported(_unit);
-                        }
-                    } else if (moduleRef != null) {
-                        Debug.Assert(moduleRef.Name.Length + 1 < name.Length, $"Expected {name} to be a child of {moduleRef.Name}");
-                        if (moduleRef.Name.Length + 1 < name.Length) {
-                            remainingParts = name.Substring(moduleRef.Name.Length + 1).Split('.');
-                        }
-                        _unit.DeclaringModule.AddModuleReference(moduleRef);
-                        return true;
-                    } else {
+                ModuleReference modRef;
+
+                bool gotAllParents = true;
+                AnalysisValue lastParent = null;
+                remainingParts = name.Split('.');
+                foreach (var part in ModulePath.GetParents(name, includeFullName: false)) {
+                    if (!ProjectState.Modules.TryImport(part, out modRef)) {
+                        gotAllParents = false;
                         break;
                     }
+                    moduleRef = modRef;
+                    (lastParent as BuiltinModule)?.AddChildModule(remainingParts[0], moduleRef.AnalysisModule);
```

Comment:
This (and the call below) are the critical part of the fix. We may have been resolving `import package.module` into two separate modules (via type stubs) where `package` didn't know about its `.module` member. So now we explicitly associate it when importing.